posthog/models/dashboard.py

```diff
@@ -104,3 +106,19 @@ def get_analytics_metadata(self) -> Dict[str, Any]:
             "has_description": self.description != "",
             "tags_count": self.tagged_items.count(),
         }
+
+
+class DashboardInsight(models.Model):
+    """
+    An explicit join model for adding insights to dashboards allows for future extension of the relationship
+    E.g. allowing the same insight to have different filters on different dashboards
```

Comment:
I guess the more immediate one is that `Insight.layouts` needs to be associated with a specific `Dashboard`. We could either put it here, or as a `Dashboard.layouts`. Is this something that will be handled in this PR or a followup?